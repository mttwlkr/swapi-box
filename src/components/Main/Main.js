import React from 'react';
import Scroll from '../Scroll/Scroll.js'
import CardContainer from '../CardContainer/CardContainer.js'
import './Main.css'

const Main = ({info, scroll, controlFunction}) => {
  return (
    <div>
      <div className="scroll-div">
      {info.length === 0 && <Scroll scroll={scroll}/>}
      </div>
      <div className="card-div">
      {info.length > 0 && <CardContainer info={info} controlFunction={controlFunction}/>}
      </div>
    </div>
  )
}

export default Main;