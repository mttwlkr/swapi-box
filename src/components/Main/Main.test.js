import React from 'react';
import { shallow, mount } from 'enzyme';
import Main from './Main.js';
import { mockCards, mockCleanFilmData } from '../../cleaners/mocks.js'

describe('Main', () => {

  let wrapper, cards, scroll, mockControlFunction 

  beforeEach(() => {
    scroll = mockCleanFilmData;
    mockControlFunction = jest.fn()
    wrapper = shallow(<Main 
      cards={mockCards} 
      scroll={mockCleanFilmData}
      controlFunction={mockControlFunction}
    />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render the cards if there are cards as props', () => {
    expect(wrapper.find('.card-div').length).toEqual(1)
    expect(wrapper.find('.scroll-div').length).toEqual(0)
  })

  it('should render the scroll if there are no cards as props', () => {
    cards = [];
    scroll = mockCleanFilmData;
    mockControlFunction = jest.fn()
    wrapper = shallow(<Main 
      cards={cards} 
      scroll={mockCleanFilmData}
      controlFunction={mockControlFunction}
    />)
    expect(wrapper.find('.scroll-div').length).toEqual(1)
    expect(wrapper.find('.card-div').length).toEqual(0)
  })
})
