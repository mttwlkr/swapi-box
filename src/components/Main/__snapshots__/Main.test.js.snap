// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    cards={
        Array [
            Object {
              "name": "Luke Skywalker",
              "planetName": "Tatooine",
              "population": "200000",
              "species": "Human",
            },
            Object {
              "name": "C-3PO",
              "planetName": "Tatooine",
              "population": "200000",
              "species": "Droid",
            },
            Object {
              "name": "R2-D2",
              "planetName": "Naboo",
              "population": "4500000000",
              "species": "Droid",
            },
            Object {
              "name": "Darth Vader",
              "planetName": "Tatooine",
              "population": "200000",
              "species": "Human",
            },
            Object {
              "name": "Leia Organa",
              "planetName": "Alderaan",
              "population": "2000000000",
              "species": "Human",
            },
            Object {
              "name": "Owen Lars",
              "planetName": "Tatooine",
              "population": "200000",
              "species": "Human",
            },
            Object {
              "name": "Beru Whitesun lars",
              "planetName": "Tatooine",
              "population": "200000",
              "species": "Human",
            },
            Object {
              "name": "R5-D4",
              "planetName": "Tatooine",
              "population": "200000",
              "species": "Droid",
            },
            Object {
              "name": "Biggs Darklighter",
              "planetName": "Tatooine",
              "population": "200000",
              "species": "Human",
            },
            Object {
              "name": "Obi-Wan Kenobi",
              "planetName": "Stewjon",
              "population": "unknown",
              "species": "Human",
            },
          ]
    }
    controlFunction={[Function]}
    scroll={
        Object {
            "date": "2002-05-16",
            "text": "There is unrest in the Galactic Senate. Several thousand solar systems have declared their intentions to leave the Republic. This separatist movement, under the leadership of the mysterious Count Dooku, has made it difficult for the limited number of Jedi Knights to maintain peace and order in the galaxy. Senator Amidala, the former Queen of Naboo, is returning to the Galactic Senate to vote on the critical issue of creating an ARMY OF THE REPUBLIC to assist the overwhelmed Jedi....",
            "title": "Attack of the Clones",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <div
          className="card-div"
>
          <CardContainer
                    cards={
                              Array [
                                        Object {
                                          "name": "Luke Skywalker",
                                          "planetName": "Tatooine",
                                          "population": "200000",
                                          "species": "Human",
                                        },
                                        Object {
                                          "name": "C-3PO",
                                          "planetName": "Tatooine",
                                          "population": "200000",
                                          "species": "Droid",
                                        },
                                        Object {
                                          "name": "R2-D2",
                                          "planetName": "Naboo",
                                          "population": "4500000000",
                                          "species": "Droid",
                                        },
                                        Object {
                                          "name": "Darth Vader",
                                          "planetName": "Tatooine",
                                          "population": "200000",
                                          "species": "Human",
                                        },
                                        Object {
                                          "name": "Leia Organa",
                                          "planetName": "Alderaan",
                                          "population": "2000000000",
                                          "species": "Human",
                                        },
                                        Object {
                                          "name": "Owen Lars",
                                          "planetName": "Tatooine",
                                          "population": "200000",
                                          "species": "Human",
                                        },
                                        Object {
                                          "name": "Beru Whitesun lars",
                                          "planetName": "Tatooine",
                                          "population": "200000",
                                          "species": "Human",
                                        },
                                        Object {
                                          "name": "R5-D4",
                                          "planetName": "Tatooine",
                                          "population": "200000",
                                          "species": "Droid",
                                        },
                                        Object {
                                          "name": "Biggs Darklighter",
                                          "planetName": "Tatooine",
                                          "population": "200000",
                                          "species": "Human",
                                        },
                                        Object {
                                          "name": "Obi-Wan Kenobi",
                                          "planetName": "Stewjon",
                                          "population": "unknown",
                                          "species": "Human",
                                        },
                                      ]
                    }
                    controlFunction={[Function]}
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <CardContainer
            cards={
                        Array [
                                    Object {
                                      "name": "Luke Skywalker",
                                      "planetName": "Tatooine",
                                      "population": "200000",
                                      "species": "Human",
                                    },
                                    Object {
                                      "name": "C-3PO",
                                      "planetName": "Tatooine",
                                      "population": "200000",
                                      "species": "Droid",
                                    },
                                    Object {
                                      "name": "R2-D2",
                                      "planetName": "Naboo",
                                      "population": "4500000000",
                                      "species": "Droid",
                                    },
                                    Object {
                                      "name": "Darth Vader",
                                      "planetName": "Tatooine",
                                      "population": "200000",
                                      "species": "Human",
                                    },
                                    Object {
                                      "name": "Leia Organa",
                                      "planetName": "Alderaan",
                                      "population": "2000000000",
                                      "species": "Human",
                                    },
                                    Object {
                                      "name": "Owen Lars",
                                      "planetName": "Tatooine",
                                      "population": "200000",
                                      "species": "Human",
                                    },
                                    Object {
                                      "name": "Beru Whitesun lars",
                                      "planetName": "Tatooine",
                                      "population": "200000",
                                      "species": "Human",
                                    },
                                    Object {
                                      "name": "R5-D4",
                                      "planetName": "Tatooine",
                                      "population": "200000",
                                      "species": "Droid",
                                    },
                                    Object {
                                      "name": "Biggs Darklighter",
                                      "planetName": "Tatooine",
                                      "population": "200000",
                                      "species": "Human",
                                    },
                                    Object {
                                      "name": "Obi-Wan Kenobi",
                                      "planetName": "Stewjon",
                                      "population": "unknown",
                                      "species": "Human",
                                    },
                                  ]
            }
            controlFunction={[Function]}
/>,
          "className": "card-div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "cards": Array [
              Object {
                "name": "Luke Skywalker",
                "planetName": "Tatooine",
                "population": "200000",
                "species": "Human",
              },
              Object {
                "name": "C-3PO",
                "planetName": "Tatooine",
                "population": "200000",
                "species": "Droid",
              },
              Object {
                "name": "R2-D2",
                "planetName": "Naboo",
                "population": "4500000000",
                "species": "Droid",
              },
              Object {
                "name": "Darth Vader",
                "planetName": "Tatooine",
                "population": "200000",
                "species": "Human",
              },
              Object {
                "name": "Leia Organa",
                "planetName": "Alderaan",
                "population": "2000000000",
                "species": "Human",
              },
              Object {
                "name": "Owen Lars",
                "planetName": "Tatooine",
                "population": "200000",
                "species": "Human",
              },
              Object {
                "name": "Beru Whitesun lars",
                "planetName": "Tatooine",
                "population": "200000",
                "species": "Human",
              },
              Object {
                "name": "R5-D4",
                "planetName": "Tatooine",
                "population": "200000",
                "species": "Droid",
              },
              Object {
                "name": "Biggs Darklighter",
                "planetName": "Tatooine",
                "population": "200000",
                "species": "Human",
              },
              Object {
                "name": "Obi-Wan Kenobi",
                "planetName": "Stewjon",
                "population": "unknown",
                "species": "Human",
              },
            ],
            "controlFunction": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <div
            className="card-div"
>
            <CardContainer
                        cards={
                                    Array [
                                                Object {
                                                  "name": "Luke Skywalker",
                                                  "planetName": "Tatooine",
                                                  "population": "200000",
                                                  "species": "Human",
                                                },
                                                Object {
                                                  "name": "C-3PO",
                                                  "planetName": "Tatooine",
                                                  "population": "200000",
                                                  "species": "Droid",
                                                },
                                                Object {
                                                  "name": "R2-D2",
                                                  "planetName": "Naboo",
                                                  "population": "4500000000",
                                                  "species": "Droid",
                                                },
                                                Object {
                                                  "name": "Darth Vader",
                                                  "planetName": "Tatooine",
                                                  "population": "200000",
                                                  "species": "Human",
                                                },
                                                Object {
                                                  "name": "Leia Organa",
                                                  "planetName": "Alderaan",
                                                  "population": "2000000000",
                                                  "species": "Human",
                                                },
                                                Object {
                                                  "name": "Owen Lars",
                                                  "planetName": "Tatooine",
                                                  "population": "200000",
                                                  "species": "Human",
                                                },
                                                Object {
                                                  "name": "Beru Whitesun lars",
                                                  "planetName": "Tatooine",
                                                  "population": "200000",
                                                  "species": "Human",
                                                },
                                                Object {
                                                  "name": "R5-D4",
                                                  "planetName": "Tatooine",
                                                  "population": "200000",
                                                  "species": "Droid",
                                                },
                                                Object {
                                                  "name": "Biggs Darklighter",
                                                  "planetName": "Tatooine",
                                                  "population": "200000",
                                                  "species": "Human",
                                                },
                                                Object {
                                                  "name": "Obi-Wan Kenobi",
                                                  "planetName": "Stewjon",
                                                  "population": "unknown",
                                                  "species": "Human",
                                                },
                                              ]
                        }
                        controlFunction={[Function]}
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <CardContainer
              cards={
                            Array [
                                          Object {
                                            "name": "Luke Skywalker",
                                            "planetName": "Tatooine",
                                            "population": "200000",
                                            "species": "Human",
                                          },
                                          Object {
                                            "name": "C-3PO",
                                            "planetName": "Tatooine",
                                            "population": "200000",
                                            "species": "Droid",
                                          },
                                          Object {
                                            "name": "R2-D2",
                                            "planetName": "Naboo",
                                            "population": "4500000000",
                                            "species": "Droid",
                                          },
                                          Object {
                                            "name": "Darth Vader",
                                            "planetName": "Tatooine",
                                            "population": "200000",
                                            "species": "Human",
                                          },
                                          Object {
                                            "name": "Leia Organa",
                                            "planetName": "Alderaan",
                                            "population": "2000000000",
                                            "species": "Human",
                                          },
                                          Object {
                                            "name": "Owen Lars",
                                            "planetName": "Tatooine",
                                            "population": "200000",
                                            "species": "Human",
                                          },
                                          Object {
                                            "name": "Beru Whitesun lars",
                                            "planetName": "Tatooine",
                                            "population": "200000",
                                            "species": "Human",
                                          },
                                          Object {
                                            "name": "R5-D4",
                                            "planetName": "Tatooine",
                                            "population": "200000",
                                            "species": "Droid",
                                          },
                                          Object {
                                            "name": "Biggs Darklighter",
                                            "planetName": "Tatooine",
                                            "population": "200000",
                                            "species": "Human",
                                          },
                                          Object {
                                            "name": "Obi-Wan Kenobi",
                                            "planetName": "Stewjon",
                                            "population": "unknown",
                                            "species": "Human",
                                          },
                                        ]
              }
              controlFunction={[Function]}
/>,
            "className": "card-div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "cards": Array [
                Object {
                  "name": "Luke Skywalker",
                  "planetName": "Tatooine",
                  "population": "200000",
                  "species": "Human",
                },
                Object {
                  "name": "C-3PO",
                  "planetName": "Tatooine",
                  "population": "200000",
                  "species": "Droid",
                },
                Object {
                  "name": "R2-D2",
                  "planetName": "Naboo",
                  "population": "4500000000",
                  "species": "Droid",
                },
                Object {
                  "name": "Darth Vader",
                  "planetName": "Tatooine",
                  "population": "200000",
                  "species": "Human",
                },
                Object {
                  "name": "Leia Organa",
                  "planetName": "Alderaan",
                  "population": "2000000000",
                  "species": "Human",
                },
                Object {
                  "name": "Owen Lars",
                  "planetName": "Tatooine",
                  "population": "200000",
                  "species": "Human",
                },
                Object {
                  "name": "Beru Whitesun lars",
                  "planetName": "Tatooine",
                  "population": "200000",
                  "species": "Human",
                },
                Object {
                  "name": "R5-D4",
                  "planetName": "Tatooine",
                  "population": "200000",
                  "species": "Droid",
                },
                Object {
                  "name": "Biggs Darklighter",
                  "planetName": "Tatooine",
                  "population": "200000",
                  "species": "Human",
                },
                Object {
                  "name": "Obi-Wan Kenobi",
                  "planetName": "Stewjon",
                  "population": "unknown",
                  "species": "Human",
                },
              ],
              "controlFunction": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
